decidim-assemblies/app/controllers/decidim/assemblies/widgets_controller.rb

```diff
@@ -5,10 +5,16 @@ module Assemblies
     class WidgetsController < Decidim::WidgetsController
       helper Decidim::SanitizeHelper
 
+      def show
+        enforce_permission_to :read, :participatory_space, current_participatory_space: model if model
+
+        super
+      end
+
       private
 
       def model
-        @model ||= Assembly.find_by(slug: params[:assembly_slug])
+        @model ||= Assembly.published.find_by(slug: params[:assembly_slug])
```

Comment:
@andreslucena , i have not really tried the code yet. But maybe is there any reason why you only used published scope here, and not something like `.public_spaces.published`? 

Do we want to expose the any widgets from private assemblies?

Reply:
Also, here we have a possible security issue when multi tenancy is involved. Reading the piece of code, it seems that is possible to access the widgets cross organization.

Reply:
> Do we want to expose the any widgets from private assemblies?

In the case of "private transparent" yes. You have the specs here: https://github.com/decidim/decidim/blob/085c50cc6e4c660176438d9a3e671772200ed561/decidim-core/lib/decidim/core/test/shared_examples/embed_resource_examples.rb#L123 

In the case of "private non transparent" no. I see that I have a bug there where if it's a private user it can access it, but it should not access it (nor have access to the link/modal for embedding). I'll work on that 

> Also, here we have a possible security issue when multi tenancy is involved. Reading the piece of code, it seems that is possible to access the widgets cross organization.
 
 I agree, I'll check that out

Reply:
> Also, here we have a possible security issue when multi tenancy is involved 

Solved with 059fd6067a

Reply:
> I see that I have a bug there where if it's a private user it can access it, but it should not access it (nor have access to the link/modal for embedding). I'll work on that

Solved with 6bff298bdd89672d75e66d81be1bf01b4310d20c